reframe/core/schedulers/slurm.py

```diff
@@ -227,7 +230,21 @@ def emit_preamble(self, job):
 
     def submit(self, job):
         cmd = f'sbatch {job.script_filename}'
-        completed = _run_strict(cmd, timeout=self._submit_timeout)
+        intervals = itertools.cycle([1, 2, 3])
+        while True:
+            try:
+                completed = _run_strict(cmd, timeout=self._submit_timeout)
+                break
+            except SpawnedProcessError as e:
+                sbatch_error = '|'.join(self._resubmit_on_qos_errors)
+                if (
+                    not self._resubmit_on_qos_errors or
+                    not re.search(sbatch_error, e.stderr)
```

Comment:
Also this is wrong. `_resubmit_on_qos_errors` is a list of QOS errors that we should ignore. So this must revised.